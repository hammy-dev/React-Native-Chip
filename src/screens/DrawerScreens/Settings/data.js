export default [
  { name: 'Change Password', screenName: 'changePassword' },
  { name: 'Change 4 Digit Pin', screenName: 'PIN' },
  { name: 'Pin Lock', toggle: true },
  { name: 'App Version', disabled: true },
  { name: 'Delete Account', screenName: 'deleteAccount' },

];
