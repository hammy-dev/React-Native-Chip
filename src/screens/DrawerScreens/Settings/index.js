import {
  View, Text, TouchableOpacity, Switch,
} from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { AntDesign } from '@expo/vector-icons';
import React, { useEffect, useState } from 'react';
import AUX from '../../../components/AUX-/index';
import data from './data';
import styles from './styles';

function Settings({ navigation }) {
  const [isEnabled, setIsEnabled] = useState(false);
  useEffect(async () => {
    const isPin = await AsyncStorage.getItem('isPinSet');
    const isPinJson = JSON.parse(isPin);
    setIsEnabled(isPinJson.isPin);
  }, [isEnabled]);
  const toggleSwitch = async () => {
    setIsEnabled((previousState) => !previousState);
    await AsyncStorage.setItem('isPinSet', JSON.stringify({ isPin: !isEnabled }));
  };
  const openView = (item) => {
    if (item.screenName) {
      navigation.navigate(item.screenName);
    }
  };
  return (
    <AUX>
      <View>
        {
          data && data.map((element) => {
            if (element.screenName || element.disabled) {
              return (
                <TouchableOpacity
                  key={element.name}
                  testID={element.screenName}
                  onPress={() => { openView(element); }}
                >
                  <View style={{
                    ...styles.innerContainer,
                  }}
                  >
                    <Text style={element.disabled
                      ? { ...styles.textStyle, ...styles.textStyleDisabled }
                      : { ...styles.textStyle }}
                    >
                      {element.name}
                    </Text>
                    <AntDesign name="right" size={24} color={element.disabled ? '#ccc' : 'black'} />
                  </View>
                </TouchableOpacity>
              );
            }

            return (
              <View
                key={element.name}
                style={{ ...styles.innerContainer }}
              >
                <Text style={{ ...styles.textStyle }}>
                  {element.name}
                </Text>
                <View style={{ ...styles.toggleStyle }}>
                  <Switch
                    testID="switch"
                    trackColor={{ false: '#767577', true: '#35c759' }}
                    thumbColor={isEnabled ? 'white' : '#f4f3f4'}
                    ios_backgroundColor="#3e3e3e"
                    onValueChange={toggleSwitch}
                    value={isEnabled}
                  />
                </View>
              </View>

            );
          })

        }
      </View>

    </AUX>
  );
}

export default Settings;
