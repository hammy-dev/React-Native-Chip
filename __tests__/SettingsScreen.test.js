import React from 'react';
import renderer, {act} from 'react-test-renderer';
import Settings from '../src/screens/DrawerScreens/Settings/index';
import {  render, fireEvent } from '@testing-library/react-native';

jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
describe('Help', () => {
    it('renders correctly', async() => {
        const mockedParams = {
            navigation: {navigate:jest.fn()}
          };
        const tree = renderer.create( <Settings {...mockedParams} />).toJSON();
        expect(tree.type).toBe( 'View')
        const component = render( <Settings {...mockedParams} />);
        const nextPress = component.getByTestId('changePassword');
        fireEvent.press(nextPress);
        let switchButton = component.getByTestId("switch")
        fireEvent(switchButton, "onValueChange", true);

    });
})