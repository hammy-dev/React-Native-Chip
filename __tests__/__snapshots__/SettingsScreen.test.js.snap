// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "#f0eefc",
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="stretch"
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="changePassword"
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderColor": "#e1e0e6",
              "borderRadius": 0,
              "elevation": 5,
              "flexDirection": "row",
              "height": "auto",
              "justifyContent": "space-between",
              "marginLeft": "5%",
              "marginTop": 20,
              "padding": 18,
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.5,
              "shadowRadius": 10,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "fontSize": 20,
                "width": "80%",
              }
            }
          >
            Change Password
          </Text>
          <Text />
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="privacyPolicy"
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderColor": "#e1e0e6",
              "borderRadius": 0,
              "elevation": 5,
              "flexDirection": "row",
              "height": "auto",
              "justifyContent": "space-between",
              "marginLeft": "5%",
              "marginTop": 20,
              "padding": 18,
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.5,
              "shadowRadius": 10,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "fontSize": 20,
                "width": "80%",
              }
            }
          >
            Change 4 Digit Pin
          </Text>
          <Text />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e1e0e6",
            "borderRadius": 0,
            "elevation": 5,
            "flexDirection": "row",
            "height": "auto",
            "justifyContent": "space-between",
            "marginLeft": "5%",
            "marginTop": 20,
            "padding": 18,
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.5,
            "shadowRadius": 10,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "fontSize": 20,
              "width": "80%",
            }
          }
        >
          Pin Lock
        </Text>
        <View
          style={
            Object {
              "height": 38,
              "marginTop": -10,
            }
          }
        >
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#e7f6f9"
            style={
              Array [
                Object {
                  "height": 31,
                  "width": 51,
                },
                Object {
                  "backgroundColor": "#3e3e3e",
                  "borderRadius": 16,
                },
              ]
            }
            testID="switch"
            thumbTintColor="#f4f3f4"
            tintColor="#767577"
            value={false}
          />
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderColor": "#e1e0e6",
              "borderRadius": 0,
              "elevation": 5,
              "flexDirection": "row",
              "height": "auto",
              "justifyContent": "space-between",
              "marginLeft": "5%",
              "marginTop": 20,
              "padding": 18,
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.5,
              "shadowRadius": 10,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#ccc",
                "fontSize": 20,
                "width": "80%",
              }
            }
          >
            App Version
          </Text>
          <Text />
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="deleteAccount"
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderColor": "#e1e0e6",
              "borderRadius": 0,
              "elevation": 5,
              "flexDirection": "row",
              "height": "auto",
              "justifyContent": "space-between",
              "marginLeft": "5%",
              "marginTop": 20,
              "padding": 18,
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.5,
              "shadowRadius": 10,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "fontSize": 20,
                "width": "80%",
              }
            }
          >
            Delete Account
          </Text>
          <Text />
        </View>
      </View>
    </View>
  </View>
</View>
`;
